caffe2/serialize/read_adapter_interface.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+namespace at {
+
+// this is the interface for the (file/stream/memory) reader in
+// PyTorchStreamReader. with this interface, we can extend the support
+// besides standard istream
+class ReadAdapterInterface {
+ public:
+  virtual size_t size() = 0;
+  virtual void seek(size_t pos) = 0;
+  virtual size_t read(void* buf, size_t size) = 0;
```

Comment:
There is no need to have separate seek/read operators. Every time the container format reads it immediately seeks before hand. I think this should just be two functions:
`size_t read(uint64_t pos, char* buf, size_t n, const char* what)` and `size_t size()`.  `validate` is weird to have separate and can be folded into read.